### PR TITLE
Initial support for Hailo-8L

### PR DESCRIPTION
**Pull request by spanner3003**
Added file for Hailo-8L detector including dockerfile, h8l.mk, h8l.hcl, hailo8l.py, ci.yml and ssd_mobilenat_v1.hef as the inference network.

Added files to help with the installation of Hailo-8L dependences like generate_wheel_conf.py, requirements-wheel-h8l.txt and modified setup.py to try and work with any hardware.

Updated docs to reflect Initial Hailo-8L support including oject_detectors.md,  hardware.md and installation.md.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -78,6 +78,16 @@ jobs:
...
+      - name: Build and push Hailo-8l build
+        uses: docker/bake-action@v4
+        with:
+          push: true
+          targets: h8l
+          files: docker/hailo8l/h8l.hcl
+          set: |
+            h8l.tags=${{ steps.setup.outputs.image-name }}-h8l
+            *.cache-from=type=registry,ref=${{ steps.setup.outputs.cache-name }}-arm64
+            *.cache-to=type=registry,ref=${{ steps.setup.outputs.cache-name }}-arm64,mode=max
```
Comment (NickM-27) [inline .github/workflows/ci.yml:90]:
```suggestion
            *.cache-from=type=registry,ref=${{ steps.setup.outputs.cache-name }}-h8l
            *.cache-to=type=registry,ref=${{ steps.setup.outputs.cache-name }}-h8l,mode=max
```

File: docs/docs/configuration/object_detectors.md [hunk 2]
```diff
@@ -393,3 +393,25 @@ $ cat /sys/kernel/debug/rknpu/load
 
 - All models are automatically downloaded and stored in the folder `config/model_cache/rknn_cache`. After upgrading Frigate, you should remove older models to free up space.
 - You can also provide your own `.rknn` model. You should not save your own models in the `rknn_cache` folder, store them directly in the `model_cache` folder or another subfolder. To convert a model to `.rknn` format see the `rknn-toolkit2` (requires a x86 machine). Note, that there is only post-processing for the supported models.
+
+## Hailo-8l
+
+This detector is availability if you are using the Raspberry pi 5 with ai kit and currently not tested if you are using the Hailo-8L with other hardware.
```
Comment (NickM-27) [inline docs/docs/configuration/object_detectors.md:399]:
```suggestion
This detector is available if you are using the Raspberry Pi 5 with Hailo-8L AI Kit. This has not been tested using the Hailo-8L with other hardware.
```

File: docs/docs/frigate/installation.md [hunk 3]
```diff
@@ -216,6 +231,7 @@ The community supported docker image tags for the current stable version are:
   - `stable-rocm-gfx900` - AMD gfx900 driver only
   - `stable-rocm-gfx1030` - AMD gfx1030 driver only
   - `stable-rocm-gfx1100` - AMD gfx1100 driver only
+- `stable-h8l` - Frigate build for the Hailo-8L M.2 PICe Raspberry pi5 hat
```
Comment (NickM-27) [inline docs/docs/frigate/installation.md:234]:
```suggestion
  - `stable-h8l` - Frigate build for the Hailo-8L M.2 PICe Raspberry pi5 hat
```

File: docs/docs/frigate/installation.md [hunk 4]
```diff
@@ -94,6 +94,21 @@ By default, the Raspberry Pi limits the amount of memory available to the GPU. I
...
+
+For Raspberry Pi 5 users with the AI Kit, installation is straightforward. Simply follow this [guide](https://www.raspberrypi.com/documentation/accessories/ai-kit.html#ai-kit-installation) to install the driver and software.
+
+For other boards, follow these steps for installation:
+
+1. Install the driver from the Hailo GitHub repository. A convenient script for Linux is available to clone the repository, build the driver, and install it.
+2. Copy or download [this script](https://gist.github.com/spanner3003/4b85751d671d4ac55f926e564f1abc3e#file-install_hailo8l_driver-sh).
+3. Ensure it has execution permissions with `sudo chmod +x install_hailo_driver.sh`.
+4. Run the script with `./install_hailo_driver.sh`.
+
```
Comment (NickM-27) [inline docs/docs/frigate/installation.md:111]: there should be a certain device that needs to be mounted into the docker container right?

**Review by NickM-27 — CHANGES_REQUESTED** [review]
This looks great! Just some comments on some things to change or clarify

What kind of inference speeds are you seeing in the system metrics page?

**Comment by hawkeye217** [comment]
Is it possible to pull `ssd_mobilenet_v1.hef` from somewhere instead of including it in the repo?

**Comment by NickM-27** [comment]
See https://github.com/blakeblackshear/frigate/blob/a4eb435f1a9f64cde8a84357ea186de9072364d8/frigate/detectors/plugins/rknn.py#L126 for an example of how rknn does it

File: docs/docs/configuration/object_detectors.md (hunk 2 above)
Comment (spanner3003) [reply docs/docs/configuration/object_detectors.md]: Thank you again. first pull request to a big repo.

**Comment by hawkeye217** [comment]
Just curious, what are the inference times you're seeing?

File: docs/docs/frigate/hardware.md [hunk 5]
```diff
@@ -107,6 +107,10 @@ Frigate supports hardware video processing on all Rockchip boards. However, hard
 
 The inference time of a rk3588 with all 3 cores enabled is typically 25-30 ms for yolo-nas s.
 
+#### Hailo-8l PCIe
+
+Frigate supports the Hailo-8l M.2 card on any harware but currently it is only tested on the Raspberry pi5 PCIe hat from the AI kit.
```
Comment (hawkeye217) [inline docs/docs/frigate/hardware.md:112]: It might be nice for users to see a typical inference time here like the other hardware above has.

Comment (hawkeye217) [inline docs/docs/frigate/hardware.md:112]:
```suggestion
Frigate supports the Hailo-8l M.2 card on any hardware but currently it is only tested on the Raspberry pi5 PCIe hat from the AI kit.
```

File: docs/docs/frigate/installation.md (hunk 4 above)
Comment (spanner3003) [reply docs/docs/frigate/installation.md:128]: yes, `/dev/hailo0` I've updated the installation.md for better clarity.

File: docs/docs/frigate/hardware.md (hunk 5 above)
Comment (spanner3003) [reply docs/docs/frigate/hardware.md]: How do I find that out?

Comment (hawkeye217) [reply docs/docs/frigate/hardware.md]: Load the frigate UI and go to the system page

File: docs/docs/frigate/hardware.md [hunk 6]
```diff
@@ -107,6 +107,12 @@ Frigate supports hardware video processing on all Rockchip boards. However, hard
 
 The inference time of a rk3588 with all 3 cores enabled is typically 25-30 ms for yolo-nas s.
 
+#### Hailo-8l PCIe
+
+Frigate supports the Hailo-8l M.2 card on any hardware but currently it is only tested on the Raspberry Pi5 PCIe hat from the AI kit.
+
+The inference time for the Hailo-8L chip at time of writing is round 17-21 ms for the SSD MobileNet Version 1.
```
Comment (hawkeye217) [inline docs/docs/frigate/hardware.md:114]:
```suggestion
The inference time for the Hailo-8L chip at time of writing is around 17-21 ms for the SSD MobileNet Version 1 model.
```

**Review by hawkeye217 — CHANGES_REQUESTED** [review]

**Comment by spanner3003** [comment]
Can I download it in the `dockerfile` like with `wget`?

**Comment by NickM-27** [comment]
It might be best to make it match rknn, but wget would still be preferable to putting it into the repo

**Comment by spanner3003** [comment]
The way rknn did it looked like they trained their own model, I don't have the know how or the time to train my own.

**Comment by NickM-27** [comment]
That's not really the relevant part. The main thing is just that the model is downloaded at runtime and store in /config/model_cache

**Comment by spanner3003** [comment]
Oh should I change where I am storing it then, I can do that.

**Comment by NickM-27** [comment]
it should be a pretty simple change to not get it inside of the Dockerfile, instead at detector startup check if the /config/model_cache/h8l/{model_name} exists, if not then download it from the repo

**Comment by NickM-27** [comment]
Actually, the reason it was done that way in rknn was due to licensing. The way you have it now is good 👍

**Comment by spanner3003** [comment]
Oh sorry missed that I've updated it to download at runtime now :D

**Comment by NickM-27** [comment]
Looks like you need to run ruff to fix formatting.

**Comment by spanner3003** [comment]
ok done. 
Is there anywhere in the docs I can write about debugging logs for hailo8l.py?

**Review by spanner3003 — COMMENTED** [review]
Made requested changes

**Comment by NickM-27** [comment]
Debugging could be done as part of the detectors documentation

**Review by NickM-27 — APPROVED** [review]

**Review by hawkeye217 — APPROVED** [review]

**Pull request merged**